src/main/scala/strawman/collection/immutable/NumericRange.scala

```diff
@@ -80,6 +82,8 @@ final class NumericRange[T](
     else locationAfterN(idx)
   }
 
+  def updated[B >: T](index: Int, elem: B): IndexedSeq[B] = collection.generic.Updated.toImmutableArray(index, elem, coll)
```

Comment:
Because it *can* be. Materializing a Range can be an expensive operation. If you want that you can call `to(IndexedSeq)` explicitly.